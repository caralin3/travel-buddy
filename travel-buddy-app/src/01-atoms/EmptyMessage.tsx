import React from 'react';
import { Link } from 'react-router-dom';
import { LinkItem } from '../types';

export interface EmptyMessageProps {
  link?: LinkItem;
  message: string;
}

export const EmptyMessage: React.FC<EmptyMessageProps> = ({ link, message }) => (
  <>
    <p className="m-0">
      <em className="h6">{message}</em>
    </p>
    {!!link && <Link to={link.path}>{link.label}</Link>}
  </>
);
