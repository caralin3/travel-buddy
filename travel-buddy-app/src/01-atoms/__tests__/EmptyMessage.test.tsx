import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { EmptyMessage } from '../EmptyMessage';

describe('EmptyMessage', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<EmptyMessage message="Message" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render with link correctly', () => {
    const wrapper = shallow(<EmptyMessage link={{ label: 'Link', path: '/' }} message="Message" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
