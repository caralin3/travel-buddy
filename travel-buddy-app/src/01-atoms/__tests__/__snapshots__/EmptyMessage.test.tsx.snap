// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyMessage should render correctly 1`] = `
<Fragment>
  <p
    className="m-0"
  >
    <em
      className="h6"
    >
      Message
    </em>
  </p>
</Fragment>
`;

exports[`EmptyMessage should render with link correctly 1`] = `
<Fragment>
  <p
    className="m-0"
  >
    <em
      className="h6"
    >
      Message
    </em>
  </p>
  <Link
    to="/"
  >
    Link
  </Link>
</Fragment>
`;
